src/sinks/aws_cloudwatch_logs/mod.rs

```diff
@@ -187,7 +187,7 @@ impl SinkConfig for CloudwatchLogsSinkConfig {
         let log_group = self.group_name.clone();
         let log_stream = self.stream_name.clone();
 
-        let client = self.create_client(cx.resolver())?;
+        let client = self.create_client(dns::Resolver)?;
```

Comment:
It doesn't need to happen as part of this PR, but it'd be nice to keep pushing this down to the point where it's actually used and get rid of the readability overhead of the threading entirely.